dpsim-models/include/dpsim-models/EMT/EMT_Ph1_Switch.h

```diff
@@ -0,0 +1,68 @@
+/* Copyright 2017-2021 Institute for Automation of Complex Power Systems,
+ *                     EONERC, RWTH Aachen University
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *********************************************************************************/
+
+#pragma once
+
+#include <dpsim-models/MNASimPowerComp.h>
+#include <dpsim-models/Solver/MNASwitchInterface.h>
+#include <dpsim-models/Solver/MNAInterface.h>
+#include <dpsim-models/Definitions.h>
+#include <dpsim-models/Logger.h>
+#include <dpsim-models/Base/Base_Ph1_Switch.h>
+
+namespace CPS {
+namespace EMT {
+namespace Ph1 {
+	/// \brief One phase EMT switch
+	///
+	/// The switch can be opened and closed.
+	/// Each state has a specific resistance value.
+	class Switch :
+		public MNASimPowerComp<Real>,
+		public Base::Ph1::Switch,
+		public SharedFactory<Switch>,
+		public MNASwitchInterface {
+
+	public:
+		/// Defines UID, name, component parameters and logging level
+		Switch(String uid, String name,	Logger::Level loglevel = Logger::Level::off);
+		/// Defines name, component parameters and logging level
+		Switch(String name, Logger::Level logLevel = Logger::Level::off)
+			: Switch(name, name, logLevel) { }
+
+		SimPowerComp<Real>::Ptr clone(String name) override;
+
+		// #### General ####
+		/// Initializes component from power flow data
+		void initializeFromNodesAndTerminals(Real frequency) override;
+
+		// #### General MNA section ####
+		void mnaCompInitialize(Real omega, Real timeStep, Attribute<Matrix>::Ptr leftVector) override;
+		/// Stamps system matrix
+		void mnaCompApplySystemMatrixStamp(SparseMatrixRow& systemMatrix) override;
+		/// Stamps right side (source) vector
+		void mnaCompApplyRightSideVectorStamp(Matrix& rightVector);
```

Comment:
You may need to either `override` or `final` for intentionality and to make clang work, see 
https://github.com/sogno-platform/dpsim/actions/runs/9782798538/job/27009957501#step:7:390